app/src/test/java/com/devmob/alaya/ui/screen/login/LoginViewmodelTest.kt

```diff
@@ -0,0 +1,152 @@
+package com.devmob.alaya.ui.screen.login
+
+import android.util.Log
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import com.devmob.alaya.data.preferences.SharedPreferences
+import com.devmob.alaya.domain.GetRoleUseCase
+import com.devmob.alaya.domain.LoginUseCase
+import com.devmob.alaya.domain.model.AuthenticationResult
+import com.devmob.alaya.domain.model.UserRole
+import io.mockk.MockKAnnotations
+import io.mockk.clearMocks
+import io.mockk.coEvery
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
+import io.mockk.mockkStatic
+import junit.framework.TestCase.assertFalse
+import junit.framework.TestCase.assertTrue
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.UnconfinedTestDispatcher
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.setMain
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class LoginViewmodelTest {
+
+    @get:Rule
+    val instantTaskExecutorRule = InstantTaskExecutorRule()
+
+    @MockK
+    private lateinit var viewModel: LoginViewModel
+
+    @MockK
+    private lateinit var loginUseCase: LoginUseCase
+
+    @MockK
+    private lateinit var getRoleUseCase: GetRoleUseCase
+
+    @MockK
+    private lateinit var prefs: SharedPreferences
+
+    private val testDispatcher = UnconfinedTestDispatcher()
+
+    @Before
+    fun setUp(){
+        MockKAnnotations.init(this, relaxed = true)
+        Dispatchers.setMain(testDispatcher)
+        viewModel = LoginViewModel(loginUseCase, getRoleUseCase, prefs)
+    }
+
+    @After
+    fun tearDown() {
+        Dispatchers.resetMain()
+        clearMocks(loginUseCase, getRoleUseCase, prefs)
+    }
+
+    @Test
+    fun `given a patient, try to login successfully and navigate to patientHomeScreen`() {
```

Comment:
Es buena práctica usar el given, when y then también en el nombre del test, así nos queda mucho más claro donde pasa cada cosa, con solo leerlo
Por ejemplo: given a patient, when try to log in correctly, then navigate to patientHomeScreen

Reply:
Intenté hacerlo en todos los tests pero me queda medio raro. forzado. Volvelo a chequear porfa